source/js/media.js

```diff
@@ -0,0 +1,81 @@
+import { addLink } from './storage.js';
+import { getDaysKey } from './date.js';
+// Popup for the media button
+const mediaImg = document.getElementById('media-img');
+const submitButton = document.getElementById('media-submit-button');
+const url = document.getElementById('media-url');
+const mediaDiv = document.getElementById('media-text-area');
+
+/**
+ * Toggle the display of the HTML element with elementID between displayType and none
+ * @param {string} elementID - The ID for the HTML element
+ * @param {string} displayType - The desired display type
+ */
+function toggle(elementID, displayType) {
+    const element = document.getElementById(elementID);
+    if (element.style.display === displayType) {
+        element.style.display = 'none';
+    } else {
+        element.style.display = displayType;
+    }
+}
+
+function youtubeUpload(link) {
+    let youtubeDiv = document.createElement('div');
+    youtubeDiv.setAttribute('class', 'media-item');
+    let youtubeIfr = document.createElement('iframe');
+    youtubeIfr.setAttribute('src', `https://www.youtube.com/embed/${link.substring(32)}`);
+    youtubeIfr.style.width = '560';
+    youtubeIfr.style.height = '315';
+    youtubeDiv.appendChild(youtubeIfr);
+    mediaDiv.appendChild(youtubeDiv);
+}
+
+function pinterestUpload(link) {
+    let pinterestDiv = document.createElement('div');
+    pinterestDiv.setAttribute('class', 'media-item');
+    let pinterestIfr = document.createElement('iframe');
```

Comment:
Codacy has a fix for the issue: ['pinterestIfr' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/cse110-sp21-group31/cse110-sp21-group31/pullRequest?prid=7577486)
```suggestion
    const pinterestIfr = document.createElement('iframe');
```